packages/manager/apps/dedicated/client/app/ip/byoip/byoip.constants.js

```diff
@@ -0,0 +1,9 @@
+export const AS_OPTIONS = ['ovh_cloud', 'own'];
+
+// reference from cloud-connect module
+export const IPV4_BLOCK_PATTERN = /^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)((\/([0-9]|[1-2][0-9]|3[0-2]))?)$/;
```

Comment:
**nitpick**:
Maybe Sonarcloud proposal to transform `[0-9]` to `/d`could be done to simplify the regec

Reply:
I think more easy 

`export const IPV4_BLOCK_PATTERN = /^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)(\.(?!$)|$)){4}$/`

Because this regex set in IPV4_BLOCK_PATTERN allows an ip in 001.001.001.001 while we do not start the ip with 0 we should allow 1.1.1.1 but not 001.001.001.001

Reply:
Please check the new regex

Reply:
The new regex autorize 3 or 4 range. 1.1.1.1 is true and 1.1.1 is true too when it should not be.